game/plugins/src/main/kotlin/gg/rsmod/plugins/content/items/ringofdueling/ring_of_dueling.plugin.kts

```diff
@@ -0,0 +1,69 @@
+package gg.rsmod.plugins.content.items.ringofdueling
+
+import gg.rsmod.plugins.content.magic.TeleportType
+import gg.rsmod.plugins.content.magic.canTeleport
+import gg.rsmod.plugins.content.magic.teleport
+
+val RING_OF_DUELING = intArrayOf(
+        Items.RING_OF_DUELING8, Items.RING_OF_DUELING7, Items.RING_OF_DUELING6,
+        Items.RING_OF_DUELING5, Items.RING_OF_DUELING4, Items.RING_OF_DUELING3,
+        Items.RING_OF_DUELING2, Items.RING_OF_DUELING1
+)
+
+private val SOUNDAREA_ID = 200
+private val SOUNDAREA_RADIUS = 5
+private val SOUNDAREA_VOLUME = 1
+
+private val LOCATIONS = mapOf(
+        "Duel Arena" to Tile(3308, 3234, 0),
+        "Castle Wars" to Tile(2440, 3089, 0),
+        "Clan Wars" to Tile(3370, 3161, 0)
+)
+
+RING_OF_DUELING.forEach { duel ->
+    LOCATIONS.forEach { location, endTile ->
+        on_equipment_option(duel, option = location) {
+            player.queue(TaskPriority.STRONG) {
+                player.teleport(endTile)
+            }
+        }
+    }
+}
+
+fun Player.teleport(endTile : Tile) {
+    if (canTeleport(TeleportType.MODERN)) {
+        if (hasEquipped(EquipmentType.RING, *RING_OF_DUELING)) {
```

Comment:
This condition should be moved up with the first condition 

`if (canTeleport(TeleportType.MODERN) && hasEquipped(EquipmentType.RING, *RING_OF_DUELING)`